lib/bolt/config/transport/docker.rb

```diff
@@ -27,10 +27,6 @@ class Docker < Base
           if @config['interpreters']
             @config['interpreters'] = normalize_interpreters(@config['interpreters'])
           end
-
-          if Bolt::Util.windows? && @config['run-as']
-            raise Bolt::ValidationError, "run-as is not supported when using PowerShell"
-          end
```

Comment:
We could potentially log a message here instead that indicates the transport does not support `run-as` on Windows and will be ignored, just in case users expect it to and are surprised. Since it would _always_ be logged when using `--run-as` on Windows, even when the transport isn't being used, it would probably want to be either at debug or trace level.

Reply:
We should be doing this validation in the PowerShell shell instead of the transports, when the shell is initialized. I still think raising an error is probably best because performing an action as a user that you don't expect could be dangerous.